node/testing/src/scenarios/multi_node/connection_discovery.rs

```diff
@@ -1,275 +1,209 @@
 use std::time::Duration;
 
-use node::{
-    event_source::Event,
-    p2p::{connection::P2pConnectionState, P2pConnectionEvent, P2pEvent, P2pPeerStatus},
-};
-use tokio::time::Instant;
-
 use crate::{
-    node::{DaemonJson, OcamlNodeTestingConfig, OcamlStep, RustNodeTestingConfig},
+    cluster::{ClusterNodeId, ClusterOcamlNodeId},
+    node::{OcamlNodeTestingConfig, RustNodeTestingConfig},
     scenario::{ListenerNode, ScenarioStep},
-    scenarios::{
-        connection_finalized_event, get_peers_iter, wait_for_connection_established, ClusterRunner,
-        Driver, PEERS_QUERY,
-    },
+    scenarios::{get_peers_iter, ClusterRunner, RunCfg, PEERS_QUERY},
+};
+use anyhow::Context;
+use node::{
+    p2p::{identify::P2pIdentifyAction, peer::P2pPeerAction, PeerId},
+    ActionKind, P2pAction,
 };
+use tokio::time::sleep;
 
 /// Ensure that Rust node can pass information about peers when used as a seed node.
+/// 1. Create rust seed node and wait for it to be ready
+/// 2. Create 2 Ocaml nodes with rust seed node as initial peer
+/// 3. Check that Ocaml nodes know each other via rust seed node
 #[derive(documented::Documented, Default, Clone, Copy)]
 pub struct RustNodeAsSeed;
 
 impl RustNodeAsSeed {
     pub async fn run(self, mut runner: ClusterRunner<'_>) {
+        std::env::set_var("OPENMINA_DISCOVERY_FILTER_ADDR", "false");
         let rust_node_id = runner.add_rust_node(RustNodeTestingConfig::devnet_default());
         let rust_node_dial_addr = runner.node(rust_node_id).unwrap().dial_addr();
+        let rust_peer_id = *rust_node_dial_addr.peer_id();
+        wait_for_node_ready(&mut runner, rust_node_id).await;
 
         let ocaml_node_config = OcamlNodeTestingConfig {
             initial_peers: vec![rust_node_dial_addr],
-            daemon_json: DaemonJson::Custom("/var/lib/coda/config_dc6bf78b.json".to_owned()),
-            block_producer: None,
+            ..Default::default()
         };
 
         let ocaml_node0 = runner.add_ocaml_node(ocaml_node_config.clone());
         let ocaml_peer_id0 = runner.ocaml_node(ocaml_node0).unwrap().peer_id();
 
+        sleep(Duration::from_secs(30)).await;
```

Comment:
@0xMimir why is this sleep here needed?

Reply:
It is to make sure node0 connect before node1.

Reply:
@0xMimir ok good, please add a comment mentioning the purpose. In the future we should have a way to encode the fact that we are waiting for some specific event to happen (with a timeout) before continuing instead of plain sleeps.